src/utils_rmw.c

```diff
@@ -143,3 +143,28 @@ human_readable_size (off_t size)
 
   return buffer;
 }
+
+/*!
+ * Verify with the user about an action. The prompt produced will be
+ * "Do you want to continue? (y/n): ". If the user provides "y",
+ * then 1 is returned, else 0 is returned.
+ * @return bool
+ */
+bool
+user_verify (void)
+{
+  printf("Do you want to continue? (y/n): ");
+  char answer;
+  bool yes = 0;
+  int char_count = 0;
+  // Go through every char to empty the buffer.
+  
+  while ((answer = getchar()) != '\n' && answer != EOF)
+  {
+    yes = (strcmp(&answer, "y")==0);
```

Comment:
This is still tripping me up a little. If I'm reading it right you're using strcmp to compare bytes at the segment of memory where answer is located at. but "answer" is of type char. So you're basically reading a segment of memory that hasn't been properly allocated, unless I'm mistaken about something.

What do you think about this:

```
	/* adapted from GNU coreutils<https://www.gnu.org/software/coreutils/> yesno.c */
	bool answer;
	  /* Test against "^[yY]", hardcoded to avoid requiring getline,
     regex, and rpmatch.  */
  int c = getchar ();
  answer = (c == 'y' || c == 'Y');
  while (c != '\n' && c != EOF)
    c = getchar ();

  return answer;
```

Reply:
> This is still tripping me up a little. If I'm reading it right you're using strcmp to compare bytes at the segment of memory where answer is located at. but "answer" is of type char. So you're basically reading a segment of memory that hasn't been properly allocated, unless I'm mistaken about something.

Actually, now that I think about it more, it should only read memory that's been "initialized" with getchar () depending on how many chars the user enters, which afaik couldn't cause a problem.

Forgive me for not understanding @Jammyjamjamman and being difficult xd Hopefully it's consolation for you to know that I'm learning from you as well. :)

I requested some additional feedback in [this LQ thread](https://www.linuxquestions.org/questions/showthread.php?p=6010952#post6010952)